src/storage/config.rs

```diff
@@ -2,27 +2,44 @@ use super::{InMemoryStorage, SqliteStorage, Storage};
 use crate::errors::Result;
 use std::path::PathBuf;
 
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub enum AccessMode {
+    ReadOnly,
+    ReadWrite,
+}
+
 /// The configuration required for a replica's storage.
+#[non_exhaustive]
 pub enum StorageConfig {
     /// Store the data on disk.  This is the common choice.
     OnDisk {
         /// Path containing the task DB.
         taskdb_dir: PathBuf,
 
-        /// Create the DB if it does not already exist
+        /// Create the DB if it does not already exist. This will occur
+        /// even if access_mode is `ReadOnly`.
         create_if_missing: bool,
+
+        /// Access mode for this database.
+        access_mode: AccessMode,
     },
     /// Store the data in memory.  This is only useful for testing.
     InMemory,
 }
 
+#[allow(deprecated)]
```

Comment:
Is this needed? From your PR comments I'm under the impression you were going to deprecate something and then changed your mind.

Reply:
Oh, thanks, I missed this one!